api/src/main/java/com/epam/pipeline/controller/resource/StaticResourcesController.java

```diff
@@ -60,23 +66,42 @@ public void getStaticFile(final HttpServletRequest request, final HttpServletRes
                     request.getPathInfo().replaceFirst(STATIC_RESOURCES, ""));
             final String fileName = FilenameUtils.getName(content.getName());
             final MediaType mediaType = getMediaType(fileName);
+            final StaticResourceSettings settings = getStaticResourceSettings(fileName);
 
-            writeStreamToResponse(response, content.getContent(), fileName, mediaType,
-                    !MediaType.APPLICATION_OCTET_STREAM.equals(mediaType), getCustomHeaders(fileName));
+            writeStreamToResponse(response, getCustomContent(content.getContent(), settings), fileName, mediaType,
+                    !MediaType.APPLICATION_OCTET_STREAM.equals(mediaType), getCustomHeaders(settings));
         } catch (InvalidPathException e) {
             response.setHeader("Location", request.getRequestURI() + ProviderUtils.DELIMITER);
             response.setStatus(HttpStatus.FOUND.value());
         }
     }
 
-    private Map<String, String> getCustomHeaders(final String fileName) {
-        final Map<String, Map<String, String>> headers = preferenceManager.getPreference(
-                SystemPreferences.DATA_SHARING_STATIC_HEADERS);
+    private StaticResourceSettings getStaticResourceSettings(final String fileName) {
+        final Map<String, StaticResourceSettings> settings = preferenceManager.getPreference(
+                SystemPreferences.DATA_SHARING_STATIC_RESOURCE_SETTINGS);
         final String extension = FilenameUtils.getExtension(fileName);
-        if (MapUtils.isEmpty(headers) || !headers.containsKey(extension)) {
-            return Collections.emptyMap();
+        return (MapUtils.isEmpty(settings) || !settings.containsKey(extension)) ? null : settings.get(extension);
```

Comment:
I think just `settings.get` will work as this - return null in case of missing value